src/libsync/propagateupload.cpp

```diff
@@ -209,30 +209,45 @@ void PropagateUploadFileQNAM::start()
 
     _stopWatch.start();
 
-    // do whatever is needed to add a checksum to the http upload request.
-    // in any case, the validator will emit signal startUpload to let the flow
-    // continue in slotStartUpload here.
-    TransmissionChecksumValidator *validator = new TransmissionChecksumValidator(filePath, this);
-
-    // If the config file does not specify a checksum type but the
-    // server supports it choose a type based on that.
-    if (validator->checksumType().isEmpty()) {
-        QStringList checksumTypes = _propagator->account()->capabilities().supportedChecksumTypes();
-        if (!checksumTypes.isEmpty()) {
-            // TODO: We might want to prefer some types over others instead
-            // of choosing the first.
-            validator->setChecksumType(checksumTypes.first());
+    auto supportedChecksumTypes = _propagator->account()->capabilities().supportedChecksumTypes();
+
+    // If we already have a checksum header and the checksum type is supported
+    // by the server, we keep that - otherwise recompute.
```

Comment:
minor: maybe add a comment where we get could possibly have the checksum header from